File: 백준/Silver/17262. 팬덤이 넘쳐흘러/팬덤이 넘쳐흘러.cc
#include <bits/stdc++.h>
using namespace std;

int main(){

    int n;

    cin >> n;



    int smallest=100001;
    int biggest=0;

    while(n--){
        int s,e;
        cin >> s >> e;

        biggest = max(s,biggest);
        smallest = min(e,smallest);

    }

    if(biggest-smallest < 0){
        cout << 0;
    }else{
        cout << biggest-smallest;
    }





}

File: 백준/Silver/17262. 팬덤이 넘쳐흘러/팬덤이 넘쳐흘러.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.math.BigInteger;
import java.util.Arrays;
import java.util.Comparator;
import java.util.StringTokenizer;
import java.util.Scanner;




public class Main {
    public static void main(String[] args) throws IOException {

        Scanner sc = new Scanner(System.in);

        int n=0;
        n=sc.nextInt();

        int biggest=0;
        int smallest=100001;

        while(n>0){

            int s,e;
            s = sc.nextInt();
            e = sc.nextInt();

            biggest = Math.max(s,biggest);
            smallest = Math.min(e,smallest);


            n--;
        }

        if(biggest-smallest < 0){
            System.out.println(0);
        }else{
            System.out.println(biggest-smallest);
        }




    }
}